public/pages/Dashboard/components/AnomaliesLiveChart.tsx

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+import { useState, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { getDetector } from '../../../redux/reducers/ad';
+import { Detector } from '../../../models/interfaces';
+import {
+  SORT_DIRECTION,
+  AD_DOC_FIELDS,
+  MIN_IN_MILLI_SECS,
+} from '../../../../server/utils/constants';
+import { searchES } from '../../../redux/reducers/elasticsearch';
+import { get } from 'lodash';
+import { getDetectorResults } from '../../../redux/reducers/anomalyResults';
+import moment, { Moment } from 'moment';
+import ContentPanel from '../../../components/ContentPanel/ContentPanel';
+import {
+  Chart,
+  Axis,
+  Settings,
+  Position,
+  getAxisId,
+  getSpecId,
+  BarSeries,
+  niceTimeFormatter,
+  ScaleType,
+  LineAnnotation,
+  AnnotationDomainTypes,
+  LineAnnotationDatum,
+} from '@elastic/charts';
+import React from 'react';
+import {
+  visualizeDetectorAnomalyResult,
+  getFloorPlotTime,
+} from '../../../../server/utils/helpers';
+import { ANOMALY_RESULT_INDEX } from '../../../utils/constants';
+
+export interface AnomaliesLiveChartProps {
+  allDetectorsSelected: boolean;
+  selectedDetectors: Detector[];
+}
+
+interface LiveTimeRangeState {
+  startDateTime: Moment;
+  endDateTime: Moment;
+}
+
+export const AnomaliesLiveChart = (props: AnomaliesLiveChartProps) => {
+  const dispatch = useDispatch();
+
+  const getRecentAnomalousDetectorsQuery = {
+    index: ANOMALY_RESULT_INDEX,
+    size: 10,
+    query: {
+      bool: {
+        must: [
+          {
+            exists: {
+              field: AD_DOC_FIELDS.ANOMALY_GRADE,
+            },
+          },
+          {
+            bool: {
+              must_not: [
+                {
+                  exists: {
+                    field: AD_DOC_FIELDS.ERROR,
+                  },
+                },
+              ],
+            },
+          },
+          {
+            range: {
+              [AD_DOC_FIELDS.ANOMALY_GRADE]: {
+                gt: 0.0,
+              },
+            },
+          },
+        ],
```

Comment:
I believe you shouldn't need the `exists` argument for the anomaly grade here. The `range` arg should be enough to only return results where anomaly grade exists AND is > 0.0. See https://github.com/ohltyler/anomaly-detection-kibana-plugin/blob/updateDetectorListPage/server/routes/utils/adHelpers.ts#L94

Reply:
good catch, will remove it.

Reply:
Also, small thing, but couldn't the `must_not` be moved outside of a nested `bool` and be in the outer `bool`?

Reply:
hmmm, good idea. I agree this can simplify the query. Will change it.